pls.h

```diff
@@ -38,8 +52,134 @@ typedef enum { LOO, NEW_DATA } VALIDATION_METHOD;
  *     a     : integer counter for latent variable dimension
  */
 
-class PLS_Model {
-  public:
+// helper methods
+template<typename MATTYPE>
+size_t find_dominant_ev (const EigenSolver<MATTYPE> es) {
+    auto eig_val = es.eigenvalues();
+    float_type m = 0;
+    size_t idx = 0;
+
+    for (size_t i = 0; i < static_cast<size_t>(eig_val.size()); i++) {
+        if (imag(eig_val[i]) == 0) {
+            if (abs(eig_val[i]) > m) {
+                m = abs(eig_val[i]);
+                idx = i;
+            }
+        }
+    }
+    return idx;
+
+};
+
+float_type dominant_eigenvalue( EigenSolver<Mat2Dc> es ){
+    const size_t idx = find_dominant_ev(es);
+    return abs(es.eigenvalues()[idx].real());
+};
+
+
+Colc dominant_eigenvector( EigenSolver<Mat2D> es ){
+    const size_t idx = find_dominant_ev(es);
+    return es.eigenvectors().col(idx);
+}
+
+#include <algorithm> // sort
+#include <numeric> // iota
+
+// tag/index sort, from https://stackoverflow.com/a/37732329/167973
+template<typename T>
+std::vector<std::size_t> ordered(const T& v)
```

Comment:
Standardize whitespace usage (not just here): `ret_type func_name (arg_type arg) {`